src/search/searcher.h

```diff
@@ -405,7 +412,14 @@ class Searcher : public std::enable_shared_from_this<Searcher> {
 
             if (score >= beta) {
                 core::TranspositionTable::writeEntry(m_stackItr->hash, score, m_stackItr->eval, move, depth, m_ply, core::TtBeta);
+
                 m_searchTables.updateKillerMoves(move, m_ply);
+
+                if (m_stackItr != m_stack.begin()) {
```

Comment:
for these checks it might be "prettier" to simply use `!isRoot` (what the rest of our checks do)